### ISSUE TITLE
Linters that operate at directory or repository level

### ISSUE THREAD

**Issue by mflova** [mfussenegger/nvim-lint#153]
Right now, one of the linters I added (`vulture`, the best linter I have found based on dead and unused variables) is adding lots of false positives.
It seems that this linter works perfectly when the full directory is provided as input, as it scans all the files where the variables might have been used. This can be the repository or even greater scope, to analyze which functions might have been used in other repositories in just less than a few seconds.
However, in `nvim-lint` you can only specify the filename when the parameter stdin is set to false, which limits the scope of some linters to analize out of the current file.

Is there any linter added that uses a greater scope than the filename and then filter all those errors to only ones that belong to the current file? I would like to implement a fix, but I am not sure where to start. I might try something in the meantime.

**Comment by mfussenegger** [comment]
You could try something like this:

```patch
diff --git a/lua/lint/linters/vulture.lua b/lua/lint/linters/vulture.lua
index 8b9579b..271d43c 100644
--- a/lua/lint/linters/vulture.lua
+++ b/lua/lint/linters/vulture.lua
@@ -3,12 +3,15 @@ local pattern = '([^:]+):(%d+):(.*)'
 local groups = { 'file', 'lnum', 'message' }

 return {
   cmd = 'vulture',
   stdin = false,
-  args = {},
+  args = function()
+    return { vim.fn.getcwd(), }
+  end,
   ignore_exitcode = true,
+  append_fname = false,
   parser = require('lint.parser').from_pattern(pattern, groups, nil, {
     ['source'] = 'vulture',
     ['severity'] = vim.diagnostic.severity.WARN,
   }),
 }
```


The linter framework by default automatically adds the filename of the buffer you're linting as argument to the command. `append_fname = false` suppresses that but then you need to add the path to the arguments manually.

The parser generated with `from_pattern` already contains logic to filter out diagnostics generated for other files - assuming there is a `file` group in the groups - which is the case for the vulture linter.

**Comment by mflova** [comment]
@mfussenegger Nice! Thanks for helping. I had no idea about that parameter. Now it is working.
I was wondering if there is a way to execute `git rev-parse --show-toplevel` to use the git repo path, and cwd if no git repo could be found. If not, I will create a PR with this fix.

**Comment by mfussenegger** [comment]
There is [io.popen](http://www.lua.org/manual/5.1/manual.html#pdf-io.popen) if you want pure lua or otherwise you could use `vim.fn.system` (see `:help system()`) this is probably easier to use in this case - as it returns the stdout as string.

**Comment by mflova** [comment]
I saw the open/popen calls but I was wondering if there was something easier. Perfect! I will open a PR later
